### PR TITLE
Handle GRACE gap

### PR DESCRIPTION
**Pull request by liviajakob**
Changes to handle gravimetry datasets which have a gap. The underlying issue here is that gravimetry datasets have a gap of 1.5 years between two missions, which needs to be handled in the algorithm.

The updated code now splits the datasets into multiple datasets at the gap and treats them individually.
For plotting and for other operations they are recombined into one dataset again (otherwise the plots are a bit confusing).

In addition the plotting functionality of gaps has been improved for the cumulative plots (now it adds extra rows to represent the gap as no data)

### CODE REVIEW

File: glambie/processing/processing_helpers.py [hunk 1]
```diff
@@ -345,6 +353,64 @@ def slice_timeseries_at_gaps(df_timeseries: pd.DataFrame) -> list[pd.DataFrame]:
...
+            errors.extend(ds.data.errors)
+        df = pd.DataFrame({"start_dates": start_dates, "end_dates": end_dates,
+                           "changes": changes, "errors": errors})
+        df = df.sort_values(by="start_dates")
+        if len(start_dates) != len(np.unique(start_dates)) or len(end_dates) != len(np.unique(end_dates)):
+            error_msg = f'''Issue with combining split datasets, duplicate dates discovered when combining:
+            {split_ds_list}'''
+            log.error(error_msg)
+            raise ValueError(error_msg)
+        new_dataset = old_datasets[0].copy()
```
Comment (OskarEarthwave) [inline glambie/processing/processing_helpers.py:396]: Could you take the 0 index or use `next` when you define it on line 378? i.e.
```old_datasets = [data_catalogue.get_filtered_catalogue(user_group=s).datasets[0] for s in split_ds_list][0]```
or
```old_datasets = next([data_catalogue.get_filtered_catalogue(user_group=s).datasets[0] for s in split_ds_list])``` :D

File: glambie/processing/processing_helpers.py [hunk 2]
```diff
@@ -345,6 +353,64 @@ def slice_timeseries_at_gaps(df_timeseries: pd.DataFrame) -> list[pd.DataFrame]:
...
+        new_dataset.data.start_dates = np.array(df["start_dates"])
+        new_dataset.data.end_dates = np.array(df["end_dates"])
+        new_dataset.data.changes = np.array(df["changes"])
+        new_dataset.data.errors = np.array(df["errors"])
+        new_dataset.user_group = new_dataset.user_group[:-2]  # remove the underscore and numbering from the name
```
Comment (OskarEarthwave) [inline glambie/processing/processing_helpers.py:401]: Optional change, this might be nicer as `new_dataset.user_group.split('_')[0]` but it's up to you

**Review by OskarEarthwave — CHANGES_REQUESTED** [review]
Looks good, only very minor comments

File: glambie/processing/processing_helpers.py [hunk 3]
```diff
@@ -276,10 +276,12 @@ def extend_annual_timeseries_if_outside_trends_period(annual_timeseries: Timeser
     return annual_timeseries_copy
 
 
-def check_and_handle_gaps_in_timeseries(data_catalogue: DataCatalogue) -> DataCatalogue:
+def check_and_handle_gaps_in_timeseries(data_catalogue: DataCatalogue) -> Tuple[DataCatalogue, list[list]]:
     """
     Checks all datasets in a timeseries if they have a temporal gap, and if so, splits them up into multiple datasets
     without gaps. If no gaps are found, the datasets stay the same.
+    User group names of the split datasets will contain have an underscore and then the sequence number appended
```
[the remark this answers is not in the document]
Comment (liviajakob) [reply glambie/processing/processing_helpers.py:283]: done

File: glambie/processing/processing_helpers.py (hunk 2 above)
Comment (liviajakob) [reply glambie/processing/processing_helpers.py:401]: I had this thought too, but the problem is that the user group names may contain underscores elsewhere

File: glambie/processing/processing_helpers.py (hunk 1 above)
Comment (liviajakob) [reply glambie/processing/processing_helpers.py:396]: This line is just used to copy the metadata of the first split dataset to fill the matadata of the newly combined dataset. We just take the first one as they should all have the same metadata, so no need to iterate to the other elements.
I realise this is not clear from the code and added a comment above to explain :)

Comment (liviajakob) [reply glambie/processing/processing_helpers.py:396]: *metadata

Comment (OskarEarthwave) [reply glambie/processing/processing_helpers.py:399]: Lovely, thanks!

**Review by OskarEarthwave — APPROVED** [review]

File: glambie/processing/processing_helpers.py (hunk 2 above)
Comment (OskarEarthwave) [reply glambie/processing/processing_helpers.py:405]: Fair enough:)

**Pull request merged**